lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -493,3 +503,66 @@ txOutMinTokenQuantity = TokenQuantity 1
 --
 txOutMaxTokenQuantity :: TokenQuantity
 txOutMaxTokenQuantity = TokenQuantity $ fromIntegral $ maxBound @Word64
+
+--------------------------------------------------------------------------------
+-- Constraints
+--------------------------------------------------------------------------------
+
+-- | Provides an abstract cost model for transactions.
+--
+-- This allows parts of a transaction to be costed individually, without
+-- having to compute the cost of an entire transaction.
+--
+data TxConstraints s = TxConstraints
+    { txBaseCost :: Coin
+      -- ^ The constant cost of an empty transaction.
+    , txBaseSize :: s
+      -- ^ The constant size of an empty transaction.
+    , txInputCost :: Coin
+      -- ^ The constant cost of a transation input.
```

Comment:
```suggestion
      -- ^ The constant cost of a transation input.
      -- 
      -- NOTE: For UTxOs sharing the same payment key this function and the current wallet in general will slightly overestimate the cost / fee, as it assumes one witness per input. (https://github.com/input-output-hk/cardano-wallet/blob/2acc0305e51d5b725701897b96cbc1ed8a61a477/lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs#L582-L584)
```

I think this deserves mentioning

Reply:
Good point!

I've added a comment in a95d51bb3726359c971ae62b92379522ff38c05e.